### ISSUE TITLE
Add support for torch.roll()

### ISSUE THREAD

**Issue by kpaigwar** [tenstorrent/tt-metal#5768]
## Requirement
While decoding Mamba model, we apply a 1D convolution filter to a cached tensor maintaining last four input states. At each timestep, this cache is shifted left and updated with new state at time t. This shift feature is implemented using torch.roll() in Pytorch model. We would ideally be requiring a kernel that efficiently updates the cache and do weighted multiplication with the 1D conv filter. 

<img width="1077" alt="Screenshot 2024-02-22 at 4 25 02 PM" src="https://github.com/tenstorrent-metal/tt-metal/assets/132708568/4601f9a7-025b-4aba-85cf-1479e67e648e">

**Comment by kpaigwar** [comment]
fyi @aguhaTT

**Comment by jliangTT** [comment]
This is assigned as P2. Does it mean it currently doesn't block the javelin work? 
Also, this looks like a tensor movement kernel. @yan-zaretskiy @TT-BrianLiu @tarafdarTT .. something in your queue?

**Comment by jliangTT** [comment]
triage into pytorch TMs for tracking for now.

**Comment by kpaigwar** [comment]
@jliangTT As we have moved to performant implementation, this Op kernel is critical. Could you please assign it to respective developers?

**Comment by jliangTT** [comment]
yes. today we are going to look at all the P1 issues that are kernel feature ask and get them assigned.

**Comment by kpaigwar** [comment]
Hey @yan-zaretskiy , is there any update on this issue? lmk if you need more information on the requirements.

**Comment by yan-zaretskiy** [comment]
Hi @kpaigwar. I'm afraid this will have to be postponed, I am working on #5428 at the moment. ETA is roughly a week.
